### DISCUSSION TITLE
Export Trace in Google Trace explorer

### DISCUSSION THREAD

**Discussion by Navirash** [strimzi/strimzi-kafka-operator#9378] [Q&A]
Hello I am reaching out to seek guidance on how to export traces using Google Trace Explorer in my strimzi configuration.  In fact, there is no specific doc.

Could you please provide me with a brief overview or steps on how to accomplish this?

**Comment by scholzj** [comment]
I never heard about Google Trace Explorer in Strimzi community. Strimzi supports OpenTelemetry for tracing, so you probably have to check with Google if they support the OpenTelemetry format / connections. I think Strimzi supports unencrypted OLTP data / connection format, right @ppatierno?

**Reply by Navirash** [reply]
So I can't export my traces in Google Trace Explorer? In the strimzi doc I saw that we use JAEGER as an exporter

**Reply by ppatierno** [reply]
I looked a little bit at Google documentation, it seems to have a service called Google Cloud Trace and they provide you an exporter library for instrumentation. You cannot use it in Strimzi. 
By using the OpenTelemetry API, you could replace the exporter using the OTEL_TRACES_EXPORTER env var but still only the ones supported by OpenTelemetry which are zipkin and otlp right now (as you can see in the official doc https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/). At the same time it would require to add the tracer jar and rebuild the image.
I think the only way you have is checking if this Cloud Trace exposes an OTLP endpoint and set that endpoint in the Strimzi configuration. Takes into account what Jakub said, there is no TLS support right now. If the Google service has TLS enabled on such an endpoint (and you cannot disable it) you would need to deploy an OpenTelemetry collector in the middle to go from non-TLS to TLS.

**Reply by Navirash** [reply]
Thank you for your reply. The only documentation I see regarding setting up OLTP in Google Trace Explorer is: https://cloud.google.com/trace/docs/otlp

I don't know if this will be compatible with strimzi. I didn't find an endpoint.

**Reply by ppatierno** [reply]
Well it's something you have to figure out by yourself I think, we don't know this Google service.
Maybe there is a way to have an OpenTelemetry collector including the Google exporter so you can "bridge" from Strimzi to the collector via OTLP and then to Google Trace. But again ... sorry, but we can't help that much.
We just explained what you would need.

**Reply by ppatierno** [reply]
I think this Ops Agent is what you could use. It sounds to me to be an OpenTelemetry collector as I was suggesting but from Google (maybe it's actually OTel collector but re-packaged by Google, don't know. The configuration  syntax looks familiar to me :-)).

https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/otlp

You could use it to bridge this way: Strimzi sends to OTLP endpoint on the Ops Agent, then the agent connects to Google Cloud Trace.

**Reply by Navirash** [reply]
Ok thank you for your help. I'm going to look into this in depth.

**Reply by ppatierno** [reply]
@Navirash I see you closed this one, can you mark what the answer was or leaving one, to help other people with the same problem please? Otherwise I will change the "category" as not Q&A.

**Reply by Navirash** [reply]
For the moment I don't know. If I find a solution I will come back to you
